didcomm/message.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Optional, List, Union, Dict
+
+from didcomm.common.types import JSON_DATA, DID, DID_URL
+
+Header = Dict[str, Union[str, int, JSON_DATA]]
+
+
+@dataclass
+class MessageOptionalHeaders:
+    """Optional headers for a message"""
+    frm: Optional[DID] = None
+    to: Optional[List[DID]] = None
```

Comment:
In what scenario would there be no "to" param? Also, it seems a bit unusual to be both `Optional` and a `List` in that a `List` could be empty

Reply:
Both `from` and `to` headers are optional according to DIDComm spec: https://identity.foundation/didcomm-messaging/spec/#message-headers
A possible scenario where there is no `to` param - a message is signed and published publicly for anyone.

And yes, `None` here means that it's not set and `to` field will not appear in the header at all.